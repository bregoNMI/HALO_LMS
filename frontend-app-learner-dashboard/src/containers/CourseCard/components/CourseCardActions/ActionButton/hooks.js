import { useWindowSize, breakpoints } from '@openedx/paragon';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  return width < breakpoints.medium.maxWidth && width > breakpoints.small.maxWidth;
};

export default useIsCollapsed;
