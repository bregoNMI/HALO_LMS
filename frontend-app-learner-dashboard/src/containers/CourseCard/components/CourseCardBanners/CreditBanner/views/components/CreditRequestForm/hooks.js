import React from 'react';

export const useCreditRequestFormData = (requestData) => {
  const ref = React.useRef(null);
  React.useEffect(() => {
    if (requestData !== null) {
      ref.current.click();
    }
  }, [requestData]);
  return { ref };
};

export default useCreditRequestFormData;
