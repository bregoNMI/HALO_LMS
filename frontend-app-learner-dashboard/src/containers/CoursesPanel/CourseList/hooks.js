import { useWindowSize, breakpoints } from '@openedx/paragon';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  return width < breakpoints.medium.maxWidth;
};

export default useIsCollapsed;
