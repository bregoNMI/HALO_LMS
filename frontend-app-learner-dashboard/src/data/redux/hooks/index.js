export * from './app';
export * from './requests';
