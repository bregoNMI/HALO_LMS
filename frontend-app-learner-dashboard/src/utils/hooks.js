import { useIntl } from '@edx/frontend-platform/i18n';

import dateFormatter from './dateFormatter';

export const useFormatDate = () => {
  const { formatDate } = useIntl();
  return (date) => dateFormatter(formatDate, date);
};

export default {
  useFormatDate,
};
