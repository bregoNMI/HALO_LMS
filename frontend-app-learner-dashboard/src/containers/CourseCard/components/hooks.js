import { reduxHooks } from 'hooks';

export const useActionDisabledState = (cardId) => {
  const { isMasquerading } = reduxHooks.useMasqueradeData();
  const {
    canUpgrade, hasAccess, isAudit, isAuditAccessExpired,
  } = reduxHooks.useCardEnrollmentData(cardId);
  const {
    isEntitlement, isFulfilled, canChange, hasSessions,
  } = reduxHooks.useCardEntitlementData(cardId);

  const { resumeUrl, homeUrl, upgradeUrl } = reduxHooks.useCardCourseRunData(cardId);

  const disableBeginCourse = !homeUrl || (isMasquerading || !hasAccess || (isAudit && isAuditAccessExpired));
  const disableResumeCourse = !resumeUrl || (isMasquerading || !hasAccess || (isAudit && isAuditAccessExpired));
  const disableViewCourse = !hasAccess || (isAudit && isAuditAccessExpired);
  const disableSelectSession = !isEntitlement || isMasquerading || !hasAccess || (!canChange || !hasSessions);
  const disableUpgradeCourse = !upgradeUrl || (isMasquerading && !canUpgrade);

  const disableCourseTitle = (isEntitlement && !isFulfilled) || disableViewCourse;

  return {
    disableBeginCourse,
    disableResumeCourse,
    disableViewCourse,
    disableUpgradeCourse,
    disableSelectSession,
    disableCourseTitle,
  };
};

export default useActionDisabledState;
