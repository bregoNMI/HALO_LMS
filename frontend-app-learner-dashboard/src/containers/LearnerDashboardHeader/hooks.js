import React from 'react';
import { useWindowSize, breakpoints } from '@openedx/paragon';
import track from 'tracking';
import { StrictDict } from 'utils';
import { linkNames } from 'tracking/constants';

import * as module from './hooks';

export const state = StrictDict({
  isOpen: (val) => React.useState(val), // eslint-disable-line
});

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  const isCollapsed = React.useMemo(() => (width <= breakpoints.large.minWidth), [width]);
  return isCollapsed;
};

export const findCoursesNavClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: linkNames.learnerHomeNavExplore,
});

export const findCoursesNavDropdownClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: linkNames.learnerHomeNavDropdownExplore,
});

export const useLearnerDashboardHeaderData = () => {
  const [isOpen, setIsOpen] = module.state.isOpen(false);
  const toggleIsOpen = () => setIsOpen(!isOpen);

  return {
    isOpen,
    toggleIsOpen,
  };
};

export default {
  useIsCollapsed,
  findCoursesNavClicked,
  findCoursesNavDropdownClicked,
  useLearnerDashboardHeaderData,
};
