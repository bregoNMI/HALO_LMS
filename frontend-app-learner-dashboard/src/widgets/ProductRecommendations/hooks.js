import { useState, useEffect } from 'react';
import { getAuthenticatedUser } from '@edx/frontend-platform/auth';

import { RequestStates, RequestKeys } from 'data/constants/requests';
import { StrictDict } from 'utils';
import { reduxHooks } from 'hooks';
import { SortKeys } from 'data/constants/app';
import { useWindowSize, breakpoints } from '@openedx/paragon';
import { useExperimentContext } from 'ExperimentContext';
import { control, treatment, noExperiment } from './constants';
import { activateProductRecommendationsExperiment, trackProductRecommendationsViewed } from './optimizelyExperiment';
import { recommendationsViewed } from './track';

import api from './api';
import * as module from './hooks';

export const state = StrictDict({
  requestState: (val) => useState(val), // eslint-disable-line
  data: (val) => useState(val), // eslint-disable-line
});

export const useIsMobile = () => {
  const { width } = useWindowSize();
  return width < breakpoints.small.minWidth;
};

export const useMostRecentCourseRunKey = () => {
  const mostRecentCourseRunKey = reduxHooks.useCurrentCourseList({
    sortBy: SortKeys.enrolled,
    filters: [],
    pageSize: 0,
  }).visible[0]?.courseRun?.courseId;

  return mostRecentCourseRunKey;
};

export const useActivateRecommendationsExperiment = () => {
  const enterpriseDashboardData = reduxHooks.useEnterpriseDashboardData();
  const hasRequestCompleted = reduxHooks.useRequestIsCompleted(RequestKeys.initialize);
  const mostRecentCourseRunKey = module.useMostRecentCourseRunKey();
  const userId = getAuthenticatedUser().userId.toString();

  const {
    experiment: { isExperimentActive },
    setExperiment,
    isMobile,
    countryCode,
  } = useExperimentContext();

  useEffect(() => {
    if (!isExperimentActive && countryCode !== null) {
      const activateExperiment = () => {
        const userAttributes = {
          is_mobile_user: isMobile,
          is_enterprise_user: !!enterpriseDashboardData,
          location: countryCode ? countryCode.toLowerCase() : '',
        };
        const experiment = activateProductRecommendationsExperiment(userId, userAttributes);

        setExperiment((prev) => ({
          ...prev,
          isExperimentActive: true,
          inRecommendationsVariant: experiment.inExperimentVariant,
        }));

        return experiment;
      };

      const sendViewedEvent = () => {
        trackProductRecommendationsViewed(userId);
        recommendationsViewed(true, control, mostRecentCourseRunKey);
      };

      if (hasRequestCompleted) {
        const { experimentActivated, inExperimentVariant } = activateExperiment();

        if (experimentActivated && !inExperimentVariant) {
          sendViewedEvent();
        }
      }
    }
  /* eslint-disable */
  }, [isExperimentActive, countryCode])
};

export const useShowRecommendationsFooter = () => {
  const { experiment } = useExperimentContext();

  return experiment;
};

export const useFetchRecommendations = (setRequestState, setData) => {
  const courseRunKey = module.useMostRecentCourseRunKey();

  useEffect(() => {
    let isMounted = true;

    const handleSuccess = (response) => {
      if (isMounted) {
        setData(response.data);
        setRequestState(RequestStates.completed);
      }
    };

    const handleError = () => {
      if (isMounted) {
        setRequestState(RequestStates.failed);
      }
    };

    if (courseRunKey) {
      api
        .fetchCrossProductRecommendations(courseRunKey)
        .then(handleSuccess)
        .catch(handleError);
    } else {
      api
        .fetchAmplitudeRecommendations()
        .then(handleSuccess)
        .catch(handleError);
    }
    return () => {
      isMounted = false;
    };
    /* eslint-disable */
  }, []);
};

export const useSendViewedEvents = (requestState, data) => {
  const mostRecentCourseRunKey = module.useMostRecentCourseRunKey();
  const userId = getAuthenticatedUser().userId.toString();

  useEffect(() => {
    if (requestState === RequestStates.completed) {
      if (data.crossProductCourses?.length === 2) {
        trackProductRecommendationsViewed(userId);
        recommendationsViewed(false, treatment, mostRecentCourseRunKey);
      } else {
        trackProductRecommendationsViewed(userId);
        recommendationsViewed(true, noExperiment, mostRecentCourseRunKey);
      }
    }
  }, [data, requestState])
}

export const useProductRecommendationsData = () => {
  const [requestState, setRequestState] = module.state.requestState(RequestStates.pending);
  const [data, setData] = module.state.data({});

  module.useFetchRecommendations(setRequestState, setData);
  module.useSendViewedEvents(requestState, data);

  return {
    productRecommendations: data,
    isLoading: requestState === RequestStates.pending,
    isLoaded: requestState === RequestStates.completed,
    hasFailed: requestState === RequestStates.failed 
  };
};

export default { useProductRecommendationsData, useShowRecommendationsFooter, useIsMobile, useActivateRecommendationsExperiment };
