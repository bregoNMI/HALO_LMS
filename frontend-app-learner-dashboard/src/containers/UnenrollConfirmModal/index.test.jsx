import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import { UnenrollConfirmModal } from '.';

import * as hooks from './hooks';

jest.mock('./components/ConfirmPane', () => 'ConfirmPane');
jest.mock('./components/ReasonPane', () => 'ReasonPane');
jest.mock('./components/FinishedPane', () => 'FinishedPane');

jest.mock('./hooks', () => ({
  __esModule: true,
  modalStates: jest.requireActual('./hooks').modalStates,
  useUnenrollData: jest.fn(),
}));

describe('UnenrollConfirmModal component', () => {
  const hookProps = {
    confirm: jest.fn().mockName('hooks.confirm'),
    reason: {
      isSkipped: false,
      reasonProps: 'other',
    },
    close: jest.fn().mockName('hooks.close'),
    closeAndRefresh: jest.fn().mockName('hooks.closeAndRefresh'),
    modalState: hooks.modalStates.confirm,
  };
  const closeModal = jest.fn().mockName('closeModal');
  const cardId = 'cardId';
  const props = {
    closeModal,
    show: true,
    cardId,
  };
  test('hooks called with closeModal and cardId', () => {
    hooks.useUnenrollData.mockReturnValueOnce(hookProps);
    shallow(<UnenrollConfirmModal {...props} />);
    expect(hooks.useUnenrollData).toHaveBeenCalledWith({ closeModal, cardId });
  });
  test('snapshot: modalStates.confirm', () => {
    hooks.useUnenrollData.mockReturnValueOnce(hookProps);
    expect(shallow(<UnenrollConfirmModal {...props} />).snapshot).toMatchSnapshot();
  });
  test('snapshot: modalStates.finished, reason given', () => {
    hooks.useUnenrollData.mockReturnValueOnce({ ...hookProps, modalState: hooks.modalStates.finished });
    expect(shallow(<UnenrollConfirmModal {...props} />).snapshot).toMatchSnapshot();
  });
  test('snapshot: modalStates.finished, reason skipped', () => {
    hooks.useUnenrollData.mockReturnValueOnce({
      ...hookProps,
      modalState: hooks.modalStates.finished,
      isSkipped: true,
    });
    expect(shallow(<UnenrollConfirmModal {...props} />).snapshot).toMatchSnapshot();
  });
  test('snapshot: modalStates.reason, should be fullscreen with no shadow', () => {
    hooks.useUnenrollData.mockReturnValueOnce({ ...hookProps, modalState: hooks.modalStates.reason });
    expect(shallow(<UnenrollConfirmModal {...props} />).snapshot).toMatchSnapshot();
  });
});
