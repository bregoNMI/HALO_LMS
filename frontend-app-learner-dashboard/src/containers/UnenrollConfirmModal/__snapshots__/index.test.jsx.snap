// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnenrollConfirmModal component snapshot: modalStates.confirm 1`] = `
<ModalDialog
  hasCloseButton={false}
  isFullscreenOnMobile={false}
  isOpen={true}
  onClose={[MockFunction utils.nullMethod]}
  title=""
>
  <div
    className="bg-white p-3 rounded shadow"
    style={
      Object {
        "textAlign": "start",
      }
    }
  >
    <ConfirmPane
      handleClose={[MockFunction hooks.close]}
      handleConfirm={[MockFunction hooks.confirm]}
    />
  </div>
</ModalDialog>
`;

exports[`UnenrollConfirmModal component snapshot: modalStates.finished, reason given 1`] = `
<ModalDialog
  hasCloseButton={false}
  isFullscreenOnMobile={false}
  isOpen={true}
  onClose={[MockFunction utils.nullMethod]}
  title=""
>
  <div
    className="bg-white p-3 rounded shadow"
    style={
      Object {
        "textAlign": "start",
      }
    }
  >
    <FinishedPane
      gaveReason={true}
      handleClose={[MockFunction hooks.closeAndRefresh]}
    />
  </div>
</ModalDialog>
`;

exports[`UnenrollConfirmModal component snapshot: modalStates.finished, reason skipped 1`] = `
<ModalDialog
  hasCloseButton={false}
  isFullscreenOnMobile={false}
  isOpen={true}
  onClose={[MockFunction utils.nullMethod]}
  title=""
>
  <div
    className="bg-white p-3 rounded shadow"
    style={
      Object {
        "textAlign": "start",
      }
    }
  >
    <FinishedPane
      gaveReason={true}
      handleClose={[MockFunction hooks.closeAndRefresh]}
    />
  </div>
</ModalDialog>
`;

exports[`UnenrollConfirmModal component snapshot: modalStates.reason, should be fullscreen with no shadow 1`] = `
<ModalDialog
  hasCloseButton={false}
  isFullscreenOnMobile={true}
  isOpen={true}
  onClose={[MockFunction utils.nullMethod]}
  title=""
>
  <div
    className="bg-white p-3 rounded"
    style={
      Object {
        "textAlign": "start",
      }
    }
  >
    <ReasonPane
      reason={
        Object {
          "isSkipped": false,
          "reasonProps": "other",
        }
      }
    />
  </div>
</ModalDialog>
`;
